### ISSUE TITLE
expose an api specific to goog.modules and their exports

### ISSUE THREAD

**Issue by MatrixFrog** [google/closure-compiler#1688]
For use by clutz. Roughly, the API should allow us to query the compiler post compilation and return the goog.modules seen, and per each the type shape of the exported var.

**Comment by ChadKillingsworth** [comment]
What's the use case? Is this goog.module specific or generalized to all module types?

**Comment by MatrixFrog** [comment]
The use case is clutz. It already gets all the goog.provides, but getting the goog.modules becomes trickier after a soon-to-be-submitted refactoring.

**Comment by MatrixFrog** [comment]
https://github.com/angular/clutz

**Comment by ChadKillingsworth** [comment]
If it gets the goog.provides prior to module rewriting (from raw source), then this would need generalized to all module types.

**Comment by MatrixFrog** [comment]
I think clutz is only concerned with goog.module for now, but you can ask the owners of that project; they may have plans related to other types of modules.

**Comment by mprobst** [comment]
@radokirov for Clutz context.

I don't think we need support for anything but `goog.module`.

**Comment by rkirov** [comment]
For goog.require/provides, we have visibility into all symbols through ```compiler.getTopScope().getAllSymbols()```

Am I missing some other type of modularity construct other than goog.module, and goog.provide/require? The only thing that comes to mind is ES6 modules, but that is not a high concern for us right now.

**Comment by Dominator008** [comment]
@rkirov There are CommonJS modules and AMD modules, but I guess they are not high concerns as well?

**Comment by ChadKillingsworth** [comment]
ES6 and CommonJS modules are rewritten to goog.require/provide calls. So as long the calls are made after module rewriting (which happens very early), then they will be covered.

**Comment by rkirov** [comment]
yep, we run as a synthetic last past, after all rewritings. Also, amd and commonjs are not a concern for our usecases.
